### ISSUE TITLE
Composed (Logical/Priority) Triggers

### ISSUE THREAD

**Issue by theauheral** [kedacore/keda#3330]
### Proposal

I would like to suggest a new feature: composed (Logical/Priority) triggers.

### Use-Case

I would like to be able to compose triggers or define a logic/priority between triggers.

Example:
Scaling from 0 to 1<=n<=maxReplicas replicas on a fixed CRON schedule (On/Off) with n managed by the metric (kafka) trigger during the runtime window.

The current design only enables "OR" logic between triggers. I would like to specify an "AND" logic between triggers or a priority of evaluation between triggers. Check trigger 1, if true then check trigger 2.

### Anything else?

Specification
- AND logic between triggers
- Alternatively, define priority triggers
- Probably most rigorous, creating a composed trigger class enabling the above two

Possible inspiration: Log4J2 rules and triggers for rotating files based on time and file size

**Comment by JorTurFer** [comment]
hey @theauheral 
Could this be duplicated? https://github.com/kedacore/keda/issues/2614

**Comment by theauheral** [comment]
Yes maybe. Different use cases but same technical implementation I suspect. I would say related but not identical.

I see "Combination of triggers can be used, either with an OR or AND fashion." which is similar yes.
but I also suggest priorities and composability. I guess #2614 has two feature requests with the second one being similar.
The example of Microsoft Azure autoscale has multiple rules but not composed rules.

**Comment by JorTurFer** [comment]
Yes, you are right.
I link this too because this is more related (https://github.com/kedacore/keda/issues/2440). 
Just to clarify this a bit, KEDA delegates the scaling to the HPA Controller so applying an AND or prioritize the metrics it's not possible. The only doable option is the creation of composite scaler IMO

**Comment by tomkerkhove** [comment]
Keep-Alive

**Comment by miqm** [comment]
@tomkerkhove any update on this?

**Comment by ghantasunil** [comment]
@tomkerkhove, any update on this?

**Comment by tomkerkhove** [comment]
Nope, are you willing to contribute it?

**Comment by zroubalik** [comment]
This is already covered by Scaling Modifiers https://keda.sh/docs/2.14/concepts/scaling-deployments/#scaling-modifiers-experimental

I belive we can close this issue already.

**Comment by joebowbeer** [comment]
@ghantasunil In theory this should be possible using a formula applied to two triggers, one based on queue length and one based on CPU.

However, as per #4992 formulas cannot be applied to a CPU trigger

**Comment by Dentrax** [comment]
Huge +1. My use case is to scale up Kafka consumers unless the error rate stays below 5%.
